cloudinit/tests/test_url_helper.py

```diff
@@ -67,6 +67,21 @@ def test_read_file_or_url_str_from_url(self):
         self.assertEqual(result.contents, data)
         self.assertEqual(str(result), data.decode('utf-8'))
 
+    @mock.patch('cloudinit.url_helper.readurl')
+    def test_read_file_or_url_passes_params_to_readurl(self, m_readurl):
```

Comment:
do we have a unittests for  1) call readurl with just url param, and then verify what the kwargs dictionary looks like (verify defaults)

Reply:
will add the unit test here too.